### PR TITLE
[MAINTENANCE] enable TRYceratops linting rules

### PR DESCRIPTION
**Pull request by JennyTee**
Enable [TRYceratops linting rules](https://docs.astral.sh/ruff/rules/#tryceratops-try) for better exception handling & raising.

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -282,6 +282,7 @@ lint.select = [
     #  https://docs.astral.sh/ruff/rules/#numpy-specific-rules-npy
     "NPY", # NumPy-specific rules TODO: Enable after fixing or ignoring errors
     "NPY201", # numpy 2.0 compatibility (see https://numpy.org/devdocs/numpy_2_0_migration_guide.html#ruff-plugin) -- Explicitly listed to view preview rule errors
+    "TRY", # TRYceratops
```
Comment (JennyTee) [inline pyproject.toml:286]: This is the main change that enables TRYceratops

**Review by Kilo59 — APPROVED** [review]
🙌 🎊

**Review by anthonyburdi — APPROVED** [review]
LGTM! Thanks for keeping the noqa additions to a single commit, much easier to review the config changes.

File: tests/ruff.toml [hunk 2]
```diff
@@ -15,6 +15,7 @@ lint.extend-ignore = [
     "PLR2004", # magic value comparison is common test pattern
     "PLR0913", # too many arguments
     "RUF015", # element index slice is common test pattern
+    "TRY003", # long messages in tests are ok
```
Comment (JennyTee) [inline tests/ruff.toml:18]: Ignoring TRY003 errors (Avoid specifying long messages outside the exception class) in test files

**Pull request merged**